src/taco-utils/commands.ts

```diff
@@ -50,14 +50,19 @@ module TacoUtility {
          */
         export interface ICommand {
             name: String;
-            run(data: ICommandData): Q.Promise<ICommandTelemetryProperties>;
+            run(args: string[]): Q.Promise<ICommandTelemetryProperties>;
             canHandleArgs(data: ICommandData): boolean;
         }
 
+        export interface ISubCommand<T extends ICommand> {
```

Comment:
Did you intend to replace instances of `ICommand` within this interface with the specific `T`?